Lib/ufo2ft/filters/explodeColorLayerGlyphs.py

```diff
@@ -0,0 +1,51 @@
+from ufo2ft.filters import BaseFilter
+from ufo2ft.util import _GlyphSet
+
+
+COLOR_LAYER_MAPPING_KEY = "com.github.googlei18n.ufo2ft.colorLayerMapping"
+COLOR_LAYERS_KEY = "com.github.googlei18n.ufo2ft.colorLayers"
+
+
+class ExplodeColorLayerGlyphsFilter(BaseFilter):
+
+    def start(self):
+        self._layerGlyphSets = {}
+
+    def set_context(self, font, glyphSet):
+        self.globalColorLayerMapping = font.lib.get(COLOR_LAYER_MAPPING_KEY)
+        font.lib[COLOR_LAYERS_KEY] = {}
+        return super().set_context(font, glyphSet)
+        
+    def _getLayer(self, font, layerName):
+        layer = self._layerGlyphSets.get(layerName)
+        if layer is None:
+            layer = _GlyphSet.from_layer(font, layerName)
+            self._layerGlyphSets[layerName] = layer
+        return layer
+
+    def filter(self, glyph):
+        """ This doesn't really filter glyphs, but copies glyphs from UFO layers
+        to alternate glyphs, for use in the COLR table.
+        """
+        font = self.context.font
+        glyphSet = self.context.glyphSet
+        colorLayers = font.lib[COLOR_LAYERS_KEY]
+        colorLayerMapping = glyph.lib.get(COLOR_LAYER_MAPPING_KEY)
+        if colorLayerMapping is None:
+            colorLayerMapping = self.globalColorLayerMapping
+        if colorLayerMapping is None:
+            # No color layer info for this glyph
+            return
+        layers = []
+        for layerName, colorID in colorLayerMapping:
+            # TODO: this does not do the right thing when
+            # the layer glyph uses components.
```

Comment:
you mean, a composite glyph in one of the color layers which references a base glyph from a different layer? UFO layers disallow this currently. Maybe we should revise.

Reply:
No, it was about using the truth value of a glyph, which (in defcon, I thought, but not sure anymore) is False if the glyph has no outlines, and doesn't check for components. But your next remark made me do it the proper way. Fixes incoming shortly!

Reply:
Wait, there was something else: composite glyphs may reference components _within the same layer_, and that is something the code currently does not handle, but should: it should copy base glyphs to the default layer with the layer name extension, and fix the references in the copied composite glyph.

Reply:
Fixed now.